<!DOCTYPE html>
<html lang="en-US" dir="ltr">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Genome Assemblers Comparison | ACRae - António Carvalho</title>
    <meta name="description" content="Antonio Carvalho CV">
    <meta name="generator" content="VitePress v1.6.3">
    <link rel="preload stylesheet" href="/assets/style.Bo4fA8YX.css" as="style">
    <link rel="preload stylesheet" href="/vp-icons.css" as="style">
    
    <script type="module" src="/assets/app.CPnjyO-B.js"></script>
    <link rel="modulepreload" href="/assets/chunks/theme.G0zfXCfi.js">
    <link rel="modulepreload" href="/assets/chunks/framework.eZ5htaLo.js">
    <link rel="modulepreload" href="/assets/projects_genome-assemblers.md.BTJKKh8Z.lean.js">
    <link rel="icon" href="/favicon.jpeg">
    <script id="check-dark-mode">(()=>{const e=localStorage.getItem("vitepress-theme-appearance")||"auto",a=window.matchMedia("(prefers-color-scheme: dark)").matches;(!e||e==="auto"?a:e==="dark")&&document.documentElement.classList.add("dark")})();</script>
    <script id="check-mac-os">document.documentElement.classList.toggle("mac",/Mac|iPhone|iPod|iPad/i.test(navigator.platform));</script>
  </head>
  <body>
    <div id="app"><div class="Layout" data-v-5d98c3a5><!--[--><!--]--><!--[--><span tabindex="-1" data-v-0b0ada53></span><a href="#VPContent" class="VPSkipLink visually-hidden" data-v-0b0ada53>Skip to content</a><!--]--><!----><header class="VPNav" data-v-5d98c3a5 data-v-ae24b3ad><div class="VPNavBar" data-v-ae24b3ad data-v-6aa21345><div class="wrapper" data-v-6aa21345><div class="container" data-v-6aa21345><div class="title" data-v-6aa21345><div class="VPNavBarTitle has-sidebar" data-v-6aa21345 data-v-1168a8e4><a class="title" href="/" data-v-1168a8e4><!--[--><!--]--><!----><span data-v-1168a8e4>ACRae - António Carvalho</span><!--[--><!--]--></a></div></div><div class="content" data-v-6aa21345><div class="content-body" data-v-6aa21345><!--[--><!--]--><div class="VPNavBarSearch search" data-v-6aa21345><!----></div><!----><!----><div class="VPNavBarAppearance appearance" data-v-6aa21345 data-v-6c893767><button class="VPSwitch VPSwitchAppearance" type="button" role="switch" title aria-checked="false" data-v-6c893767 data-v-5337faa4 data-v-1d5665e3><span class="check" data-v-1d5665e3><span class="icon" data-v-1d5665e3><!--[--><span class="vpi-sun sun" data-v-5337faa4></span><span class="vpi-moon moon" data-v-5337faa4></span><!--]--></span></span></button></div><div class="VPSocialLinks VPNavBarSocialLinks social-links" data-v-6aa21345 data-v-0394ad82 data-v-7bc22406><!--[--><a class="VPSocialLink no-icon" href="https://github.com/ACRae" aria-label="github" target="_blank" rel="noopener" data-v-7bc22406 data-v-bd121fe5><span class="vpi-social-github"></span></a><!--]--></div><div class="VPFlyout VPNavBarExtra extra" data-v-6aa21345 data-v-bb2aa2f0 data-v-cf11d7a2><button type="button" class="button" aria-haspopup="true" aria-expanded="false" aria-label="extra navigation" data-v-cf11d7a2><span class="vpi-more-horizontal icon" data-v-cf11d7a2></span></button><div class="menu" data-v-cf11d7a2><div class="VPMenu" data-v-cf11d7a2 data-v-b98bc113><!----><!--[--><!--[--><!----><div class="group" data-v-bb2aa2f0><div class="item appearance" data-v-bb2aa2f0><p class="label" data-v-bb2aa2f0>Appearance</p><div class="appearance-action" data-v-bb2aa2f0><button class="VPSwitch VPSwitchAppearance" type="button" role="switch" title aria-checked="false" data-v-bb2aa2f0 data-v-5337faa4 data-v-1d5665e3><span class="check" data-v-1d5665e3><span class="icon" data-v-1d5665e3><!--[--><span class="vpi-sun sun" data-v-5337faa4></span><span class="vpi-moon moon" data-v-5337faa4></span><!--]--></span></span></button></div></div></div><div class="group" data-v-bb2aa2f0><div class="item social-links" data-v-bb2aa2f0><div class="VPSocialLinks social-links-list" data-v-bb2aa2f0 data-v-7bc22406><!--[--><a class="VPSocialLink no-icon" href="https://github.com/ACRae" aria-label="github" target="_blank" rel="noopener" data-v-7bc22406 data-v-bd121fe5><span class="vpi-social-github"></span></a><!--]--></div></div></div><!--]--><!--]--></div></div></div><!--[--><!--]--><button type="button" class="VPNavBarHamburger hamburger" aria-label="mobile navigation" aria-expanded="false" aria-controls="VPNavScreen" data-v-6aa21345 data-v-e5dd9c1c><span class="container" data-v-e5dd9c1c><span class="top" data-v-e5dd9c1c></span><span class="middle" data-v-e5dd9c1c></span><span class="bottom" data-v-e5dd9c1c></span></span></button></div></div></div></div><div class="divider" data-v-6aa21345><div class="divider-line" data-v-6aa21345></div></div></div><!----></header><div class="VPLocalNav has-sidebar empty" data-v-5d98c3a5 data-v-a6f0e41e><div class="container" data-v-a6f0e41e><button class="menu" aria-expanded="false" aria-controls="VPSidebarNav" data-v-a6f0e41e><span class="vpi-align-left menu-icon" data-v-a6f0e41e></span><span class="menu-text" data-v-a6f0e41e>Menu</span></button><div class="VPLocalNavOutlineDropdown" style="--vp-vh:0px;" data-v-a6f0e41e data-v-8a42e2b4><button data-v-8a42e2b4>Return to top</button><!----></div></div></div><aside class="VPSidebar" data-v-5d98c3a5 data-v-319d5ca6><div class="curtain" data-v-319d5ca6></div><nav class="nav" id="VPSidebarNav" aria-labelledby="sidebar-aria-label" tabindex="-1" data-v-319d5ca6><span class="visually-hidden" id="sidebar-aria-label" data-v-319d5ca6> Sidebar Navigation </span><!--[--><!--]--><!--[--><div class="no-transition group" data-v-c40bc020><section class="VPSidebarItem level-0 has-active" data-v-c40bc020 data-v-b3fd67f8><div class="item" role="button" tabindex="0" data-v-b3fd67f8><div class="indicator" data-v-b3fd67f8></div><h2 class="text" data-v-b3fd67f8>Projects</h2><!----></div><div class="items" data-v-b3fd67f8><!--[--><div class="VPSidebarItem level-1 is-link" data-v-b3fd67f8 data-v-b3fd67f8><div class="item" data-v-b3fd67f8><div class="indicator" data-v-b3fd67f8></div><a class="VPLink link link" href="/projects/postchat.html" data-v-b3fd67f8><!--[--><p class="text" data-v-b3fd67f8>PostChat</p><!--]--></a><!----></div><!----></div><div class="VPSidebarItem level-1 is-link" data-v-b3fd67f8 data-v-b3fd67f8><div class="item" data-v-b3fd67f8><div class="indicator" data-v-b3fd67f8></div><a class="VPLink link link" href="/projects/tauri-starter.html" data-v-b3fd67f8><!--[--><p class="text" data-v-b3fd67f8>Tauri Starter</p><!--]--></a><!----></div><!----></div><div class="VPSidebarItem level-1 is-link" data-v-b3fd67f8 data-v-b3fd67f8><div class="item" data-v-b3fd67f8><div class="indicator" data-v-b3fd67f8></div><a class="VPLink link link" href="/projects/genome-assemblers.html" data-v-b3fd67f8><!--[--><p class="text" data-v-b3fd67f8>Genome Assemblers Comparison</p><!--]--></a><!----></div><!----></div><div class="VPSidebarItem level-1 is-link" data-v-b3fd67f8 data-v-b3fd67f8><div class="item" data-v-b3fd67f8><div class="indicator" data-v-b3fd67f8></div><a class="VPLink link link" href="/projects/mobile-cmunext.html" data-v-b3fd67f8><!--[--><p class="text" data-v-b3fd67f8>Mobile-CMUNeXt</p><!--]--></a><!----></div><!----></div><!--]--></div></section></div><!--]--><!--[--><!--]--></nav></aside><div class="VPContent has-sidebar" id="VPContent" data-v-5d98c3a5 data-v-1428d186><div class="VPDoc has-sidebar has-aside" data-v-1428d186 data-v-39a288b8><!--[--><!--]--><div class="container" data-v-39a288b8><div class="aside" data-v-39a288b8><div class="aside-curtain" data-v-39a288b8></div><div class="aside-container" data-v-39a288b8><div class="aside-content" data-v-39a288b8><div class="VPDocAside" data-v-39a288b8 data-v-3f215769><!--[--><!--]--><!--[--><!--]--><nav aria-labelledby="doc-outline-aria-label" class="VPDocAsideOutline" data-v-3f215769 data-v-a5bbad30><div class="content" data-v-a5bbad30><div class="outline-marker" data-v-a5bbad30></div><div aria-level="2" class="outline-title" id="doc-outline-aria-label" role="heading" data-v-a5bbad30>On this page</div><ul class="VPDocOutlineItem root" data-v-a5bbad30 data-v-b933a997><!--[--><!--]--></ul></div></nav><!--[--><!--]--><div class="spacer" data-v-3f215769></div><!--[--><!--]--><!----><!--[--><!--]--><!--[--><!--]--></div></div></div></div><div class="content" data-v-39a288b8><div class="content-container" data-v-39a288b8><!--[--><!--]--><main class="main" data-v-39a288b8><div style="position:relative;" class="vp-doc _projects_genome-assemblers" data-v-39a288b8><div><h1 id="genome-assemblers-comparison" tabindex="-1">Genome Assemblers Comparison <a class="header-anchor" href="#genome-assemblers-comparison" aria-label="Permalink to &quot;Genome Assemblers Comparison&quot;">​</a></h1><p>A Docker image based workflow for the comparison of several genome assemblers.</p><h2 id="intro" tabindex="-1">Intro <a class="header-anchor" href="#intro" aria-label="Permalink to &quot;Intro&quot;">​</a></h2><p>Genome assembly is a computational representation of a genome sequence and it is a pivotal step in genome analysis. Given the limitations of current sequencing technologies, where it&#39;s impractical to sequence entire chromosomes, genome assembly becomes essential for studying genetic structures comprehensively.</p><p>Genome reconstruction is possible due to the existence of different sequence assemblers that use different algorithms to achieve this reconstruction, each with its advantages and limitations. These technologies break the genome into smaller fragments and sequence these fragments individually. These sequenced fragments are known as &quot;reads&quot;.</p><p>Among these algorithms, De Bruijn graph-based approaches have gained prominence due to their computational efficiency and ability to manage large volumes of data. These algorithms segment the DNA sequences into overlapping k-mers, forming a graph where nodes represent k-mers and edges represent the overlaps between them.</p><p>The workflow is setup to compare the following assemblers:</p><ul><li><a href="https://github.com/bcgsc/abyss" target="_blank" rel="noreferrer">ABySS</a></li><li><a href="https://github.com/trinityrnaseq/trinityrnaseq" target="_blank" rel="noreferrer">Trinity</a></li><li><a href="https://github.com/ablab/spades" target="_blank" rel="noreferrer">SPAdes</a></li></ul><h2 id="requirements" tabindex="-1">Requirements <a class="header-anchor" href="#requirements" aria-label="Permalink to &quot;Requirements&quot;">​</a></h2><ul><li>Docker</li><li>Python 3</li></ul><h2 id="instructions" tabindex="-1">Instructions <a class="header-anchor" href="#instructions" aria-label="Permalink to &quot;Instructions&quot;">​</a></h2><h3 id="_1-build-image-and-run-docker-container" tabindex="-1">1. Build image and run docker container <a class="header-anchor" href="#_1-build-image-and-run-docker-container" aria-label="Permalink to &quot;1. Build image and run docker container&quot;">​</a></h3><ul><li>Windows: Run the <code>./run_container.bat</code></li><li>Linux &amp; MacOS: Run the <code>./run_container.sh</code></li></ul><p>This will build the docker image and automatically run the container in the terminal.</p><p>You can use this script to open the container again.</p><h3 id="_2-get-the-sra-data" tabindex="-1">2. Get the SRA data <a class="header-anchor" href="#_2-get-the-sra-data" aria-label="Permalink to &quot;2. Get the SRA data&quot;">​</a></h3><p>In the container terminal run:</p><ul><li>1st <code>./prefetch_data.sh</code></li><li>2nd <code>./fasterq_dump.sh</code></li></ul><p>You will find the <code>cache</code> and <code>fastq</code> folders in your directory.</p><p>You will only need to do this once!</p><h3 id="_3-run-the-assemblers" tabindex="-1">3. Run the assemblers <a class="header-anchor" href="#_3-run-the-assemblers" aria-label="Permalink to &quot;3. Run the assemblers&quot;">​</a></h3><p>In order to run all the assemblers all you have to do is:</p><p><code>run_all &lt;path to the fastq folder&gt; &lt;RAM usage&gt;</code></p><p>Example: <code>run_all fastq 16G</code></p><p>This will start the genome assembly for every data sample.</p><h3 id="_4-evaluate-the-contigs" tabindex="-1">4. Evaluate the contigs <a class="header-anchor" href="#_4-evaluate-the-contigs" aria-label="Permalink to &quot;4. Evaluate the contigs&quot;">​</a></h3><p>Keep in mind that some assemblies might fail without warning and so some evaluations are not present.</p><p>The evaluation is done using the Quast tool <code>https://quast.sourceforge.net/</code>.</p><p>To evaluate the model run <code>./quast_eval.sh</code></p><h3 id="_5-plot-monitor-graphs-experimental" tabindex="-1">5. Plot monitor graphs (experimental) <a class="header-anchor" href="#_5-plot-monitor-graphs-experimental" aria-label="Permalink to &quot;5. Plot monitor graphs (experimental)&quot;">​</a></h3><p>A monitor program was implemented to keep track of the RAM memory usage and the CPU usage. It is experimental and the results tend to vary.</p><p>This step attempts to plot the RAM and CPU usage over time for each assembly tool run.</p><p>Install python requirements:</p><ul><li><code>pip install -r requirements.txt [--break-system-packages]</code></li></ul><p>Get the geckodriver and setup PATH:</p><ul><li><a href="https://github.com/mozilla/geckodriver/releases" target="_blank" rel="noreferrer">https://github.com/mozilla/geckodriver/releases</a></li></ul><p>Plot graphs:</p><ul><li>Run: <code>python3 plot_monitor.py</code></li></ul><h2 id="results" tabindex="-1">Results <a class="header-anchor" href="#results" aria-label="Permalink to &quot;Results&quot;">​</a></h2><p>The experiment results can be found here <a href="https://drive.google.com/drive/folders/1spAZ4o9QbmJz_wFbOouq3-vzsJaqpFvK?usp=sharing" target="_blank" rel="noreferrer">Evaluation and Assembly Output</a></p></div></div></main><footer class="VPDocFooter" data-v-39a288b8 data-v-e257564d><!--[--><!--]--><!----><nav class="prev-next" aria-labelledby="doc-footer-aria-label" data-v-e257564d><span class="visually-hidden" id="doc-footer-aria-label" data-v-e257564d>Pager</span><div class="pager" data-v-e257564d><a class="VPLink link pager-link prev" href="/projects/tauri-starter.html" data-v-e257564d><!--[--><span class="desc" data-v-e257564d>Previous page</span><span class="title" data-v-e257564d>Tauri Starter</span><!--]--></a></div><div class="pager" data-v-e257564d><a class="VPLink link pager-link next" href="/projects/mobile-cmunext.html" data-v-e257564d><!--[--><span class="desc" data-v-e257564d>Next page</span><span class="title" data-v-e257564d>Mobile-CMUNeXt</span><!--]--></a></div></nav></footer><!--[--><!--]--></div></div></div><!--[--><!--]--></div></div><!----><!--[--><!--]--></div></div>
    <script>window.__VP_HASH_MAP__=JSON.parse("{\"index.md\":\"BumJOoKZ\",\"projects_genome-assemblers.md\":\"BTJKKh8Z\",\"projects_mobile-cmunext.md\":\"C_YG5bbK\",\"projects_postchat.md\":\"DJdhguZX\",\"projects_tauri-starter.md\":\"DTg9GTf6\"}");window.__VP_SITE_DATA__=JSON.parse("{\"lang\":\"en-US\",\"dir\":\"ltr\",\"title\":\"ACRae - António Carvalho\",\"description\":\"Antonio Carvalho CV\",\"base\":\"/\",\"head\":[],\"router\":{\"prefetchLinks\":true},\"appearance\":true,\"themeConfig\":{\"sidebar\":[{\"text\":\"Projects\",\"items\":[{\"text\":\"PostChat\",\"link\":\"/projects/postchat\"},{\"text\":\"Tauri Starter\",\"link\":\"/projects/tauri-starter\"},{\"text\":\"Genome Assemblers Comparison\",\"link\":\"/projects/genome-assemblers\"},{\"text\":\"Mobile-CMUNeXt\",\"link\":\"/projects/mobile-cmunext\"}]}],\"socialLinks\":[{\"icon\":\"github\",\"link\":\"https://github.com/ACRae\"}]},\"locales\":{},\"scrollOffset\":134,\"cleanUrls\":false}");</script>
    
  </body>
</html>