import{_ as a,c as t,o as r,ae as o}from"./chunks/framework.eZ5htaLo.js";const d=JSON.parse('{"title":"Genome Assemblers Comparison","description":"","frontmatter":{},"headers":[],"relativePath":"projects/genome-assemblers.md","filePath":"projects/genome-assemblers.md"}'),s={name:"projects/genome-assemblers.md"};function n(i,e,l,h,m,c){return r(),t("div",null,e[0]||(e[0]=[o('<h1 id="genome-assemblers-comparison" tabindex="-1">Genome Assemblers Comparison <a class="header-anchor" href="#genome-assemblers-comparison" aria-label="Permalink to &quot;Genome Assemblers Comparison&quot;">​</a></h1><p>A Docker image based workflow for the comparison of several genome assemblers.</p><h2 id="intro" tabindex="-1">Intro <a class="header-anchor" href="#intro" aria-label="Permalink to &quot;Intro&quot;">​</a></h2><p>Genome assembly is a computational representation of a genome sequence and it is a pivotal step in genome analysis. Given the limitations of current sequencing technologies, where it&#39;s impractical to sequence entire chromosomes, genome assembly becomes essential for studying genetic structures comprehensively.</p><p>Genome reconstruction is possible due to the existence of different sequence assemblers that use different algorithms to achieve this reconstruction, each with its advantages and limitations. These technologies break the genome into smaller fragments and sequence these fragments individually. These sequenced fragments are known as &quot;reads&quot;.</p><p>Among these algorithms, De Bruijn graph-based approaches have gained prominence due to their computational efficiency and ability to manage large volumes of data. These algorithms segment the DNA sequences into overlapping k-mers, forming a graph where nodes represent k-mers and edges represent the overlaps between them.</p><p>The workflow is setup to compare the following assemblers:</p><ul><li><a href="https://github.com/bcgsc/abyss" target="_blank" rel="noreferrer">ABySS</a></li><li><a href="https://github.com/trinityrnaseq/trinityrnaseq" target="_blank" rel="noreferrer">Trinity</a></li><li><a href="https://github.com/ablab/spades" target="_blank" rel="noreferrer">SPAdes</a></li></ul><h2 id="requirements" tabindex="-1">Requirements <a class="header-anchor" href="#requirements" aria-label="Permalink to &quot;Requirements&quot;">​</a></h2><ul><li>Docker</li><li>Python 3</li></ul><h2 id="instructions" tabindex="-1">Instructions <a class="header-anchor" href="#instructions" aria-label="Permalink to &quot;Instructions&quot;">​</a></h2><h3 id="_1-build-image-and-run-docker-container" tabindex="-1">1. Build image and run docker container <a class="header-anchor" href="#_1-build-image-and-run-docker-container" aria-label="Permalink to &quot;1. Build image and run docker container&quot;">​</a></h3><ul><li>Windows: Run the <code>./run_container.bat</code></li><li>Linux &amp; MacOS: Run the <code>./run_container.sh</code></li></ul><p>This will build the docker image and automatically run the container in the terminal.</p><p>You can use this script to open the container again.</p><h3 id="_2-get-the-sra-data" tabindex="-1">2. Get the SRA data <a class="header-anchor" href="#_2-get-the-sra-data" aria-label="Permalink to &quot;2. Get the SRA data&quot;">​</a></h3><p>In the container terminal run:</p><ul><li>1st <code>./prefetch_data.sh</code></li><li>2nd <code>./fasterq_dump.sh</code></li></ul><p>You will find the <code>cache</code> and <code>fastq</code> folders in your directory.</p><p>You will only need to do this once!</p><h3 id="_3-run-the-assemblers" tabindex="-1">3. Run the assemblers <a class="header-anchor" href="#_3-run-the-assemblers" aria-label="Permalink to &quot;3. Run the assemblers&quot;">​</a></h3><p>In order to run all the assemblers all you have to do is:</p><p><code>run_all &lt;path to the fastq folder&gt; &lt;RAM usage&gt;</code></p><p>Example: <code>run_all fastq 16G</code></p><p>This will start the genome assembly for every data sample.</p><h3 id="_4-evaluate-the-contigs" tabindex="-1">4. Evaluate the contigs <a class="header-anchor" href="#_4-evaluate-the-contigs" aria-label="Permalink to &quot;4. Evaluate the contigs&quot;">​</a></h3><p>Keep in mind that some assemblies might fail without warning and so some evaluations are not present.</p><p>The evaluation is done using the Quast tool <code>https://quast.sourceforge.net/</code>.</p><p>To evaluate the model run <code>./quast_eval.sh</code></p><h3 id="_5-plot-monitor-graphs-experimental" tabindex="-1">5. Plot monitor graphs (experimental) <a class="header-anchor" href="#_5-plot-monitor-graphs-experimental" aria-label="Permalink to &quot;5. Plot monitor graphs (experimental)&quot;">​</a></h3><p>A monitor program was implemented to keep track of the RAM memory usage and the CPU usage. It is experimental and the results tend to vary.</p><p>This step attempts to plot the RAM and CPU usage over time for each assembly tool run.</p><p>Install python requirements:</p><ul><li><code>pip install -r requirements.txt [--break-system-packages]</code></li></ul><p>Get the geckodriver and setup PATH:</p><ul><li><a href="https://github.com/mozilla/geckodriver/releases" target="_blank" rel="noreferrer">https://github.com/mozilla/geckodriver/releases</a></li></ul><p>Plot graphs:</p><ul><li>Run: <code>python3 plot_monitor.py</code></li></ul><h2 id="results" tabindex="-1">Results <a class="header-anchor" href="#results" aria-label="Permalink to &quot;Results&quot;">​</a></h2><p>The experiment results can be found here <a href="https://drive.google.com/drive/folders/1spAZ4o9QbmJz_wFbOouq3-vzsJaqpFvK?usp=sharing" target="_blank" rel="noreferrer">Evaluation and Assembly Output</a></p>',40)]))}const p=a(s,[["render",n]]);export{d as __pageData,p as default};
